bgpd/bgp_vty.c

```diff
@@ -10471,6 +10471,28 @@ static void bgp_show_failed_summary(struct vty *vty, struct bgp *bgp,
 	}
 }
 
+/* If the peer's description includes whitespaces
+ * then return the first occurrence. Also strip description
+ * to the given size if needed.
+ */
+static char *bgp_peer_description_stripped(char *desc, unsigned int size)
+{
+	char *stripped;
+	char *pnt;
+	unsigned int len = size > strlen(desc) ? strlen(desc) : size;
+
+	pnt = strchr(desc, ' ');
+	if (pnt)
+		len = size > (unsigned int)(pnt - desc)
+			      ? (unsigned int)(pnt - desc)
+			      : size;
+
+	stripped = XMALLOC(MTYPE_TMP, len + 1);
+	memcpy(stripped, desc, len);
```

Comment:
I Still see XMALLOC.
can it be avoided ?

Reply:
How would you prefer in this case?

Reply:
I did not think about it.
a static max string of 255 bytes?

Reply:
Changed.